packages/core/src/models/uischema.ts

```diff
@@ -25,17 +25,42 @@
 
 import { JsonSchema } from './jsonSchema';
 
+/**
+ * Interface for describing an UI schema element that may reference
+ * a subschema. The value of the scope may be a JSON Pointer or null.
+ */
+export interface Scopable {
+  /**
+   * The scope that determines to which part this element may be bound to.
+   */
+  scope?: string;
+}
+
```

Comment:
If `null` is a foreseen value then this should be reflected in the typings, i.e. `scope?: string | null`. However we don't have any other nullebale types in JSON Forms, so I'm a bit hesitant to introduce this now. Is there a specific use case you have in mind?

This new interface with the optional scope is only used in one place: The new replacement for `Scopable` called `Scoped`. What is the use case of an element with optional scope?

Reply:
The case for this change would be the option to add scopes to existing classes without breaking the API in the future. This allows adding the interface to Categorization and Category in a second step to enable relative scopes in their children.